backend/inspirehep/pidstore/providers/bai.py

```diff
@@ -78,6 +78,9 @@ def next_bai_number(cls, bai):
         return int(last_number) + 1
 
     @classmethod
+    @backoff.on_exception(
+        backoff.expo, (IntegrityError, PIDAlreadyExistsError), max_tries=5
```

Comment:
I think the point is that you don't need exponential backoff in this case. AFAIU, the failure mode is the following: there's a concurrent transaction that has created the next BAI with the same prefix already, so the current transaction comptues the same BAI, which fails when inserting as that PID already exists. But this means that the other transaction has already been committed (otherwise there would be no issue, this transaction would win), so you don't need to wait before retrying.

Reply:
But how it's possible that it was failing with current approach? We were retrying 5 times with `time.sleep(5)` for every retry and still we were getting that error.

Reply:
ok, I looked some more and I believe this is the reason: https://github.com/inspirehep/inspirehep/blob/2f0da315b970425b012467f3698343b17a21c0a6/backend/inspirehep/records/api/base.py#L472.
In principle, newly created/modified model instances should be flushed to the DB before querying so they're available to the DB. If they're not flushed, they're simply not present in the DB (they exist only in the Python interpreter), so can't be queried. See also this excelled answer https://stackoverflow.com/a/4202016.

So I guess the failure mode is not a concurrent transaction, but a different BAI created in the same transaction, which isn't flushed to the DB yet.

Question: do we really need to disable autoflush? it was done by @pazembrz in https://github.com/inspirehep/inspirehep/pull/1727 while trying to fix the versioning issues, but I'm not sure it's actually needed.

Reply:
It seems very reasonable, but this is only in update method. I think that this problem exists also on create (because we're creating same author in the concurrent transactions)

Reply:
still, *exponential backoff* doesn't make a lot of sense to me. Why not have a constant backoff insteand?

Reply:
Make sense to me too, what value do you suggest in that case?

Reply:
5s like previously?